test_libs/pyspec/eth2spec/utils/ssz/ssz_typing.py

```diff
@@ -208,221 +138,115 @@ def __new__(cls, class_name, parents, attrs):
 
     def __getitem__(self, params):
         if not isinstance(params, tuple) or len(params) != 2:
-            raise Exception("Vector must be instantiated with two args: elem type and length")
-        o = self.__class__(self.__name__, (Vector,), {'elem_type': params[0], 'length': params[1]})
-        o._name = 'Vector'
+            raise Exception("List must be instantiated with two args: elem type and length")
+        o = self.__class__(self.__name__, (self,), {'elem_type': params[0], 'length': params[1]})
+        o._name = 'AbstractList'
         return o
 
-    def __subclasscheck__(self, sub):
-        return _is_vector_instance_of(self, sub)
-
-    def __instancecheck__(self, other):
-        return _is_vector_instance_of(self, other.__class__)
-
-    def __eq__(self, other):
-        return _is_equal_vector_type(self, other)
-
-    def __ne__(self, other):
-        return not _is_equal_vector_type(self, other)
-
-    def __hash__(self):
-        return hash(self.__class__)
-
+    def __instancecheck__(self, obj):
+        if obj.__class__.__name__ != self.__name__:
+            return False
+        if hasattr(self, 'elem_type') and obj.__class__.elem_type != self.elem_type:
+            return False
+        if hasattr(self, 'length') and obj.__class__.length != self.length:
+            return False
+        return True
 
-class Vector(metaclass=VectorMeta):
+class ValueCheckError(Exception):
+    pass
+
+class AbstractList(metaclass=AbstractListMeta):
```

Comment:
If we're not using `AbstractList` directly, we shouldn't need to create an `AbstractListMeta` for it, we can leave it out. (or maybe leave it out in the subclass declarations, but unsure if that breaks things)